src/services/CertificateGenerationService.ts

```diff
@@ -177,6 +171,39 @@ class CertificateGenerationService {
             });
     }
 
+    /**
+     * Method to check if welsh feature flags are enabled and check if test station has a Welsh postcode
+     * @param testResult
+     * @param flags
+     * @returns boolean for if test station is Welsh
+     */
+    public async isTestStationWelsh(testResult: any, flags: any): Promise<boolean> {
```

Comment:
I like the simplified logic, and would be keen on everyone elses thoughts, but is there any chance you can bust the 'shouldTranslate' logic out into a separate method? To make it a bit more readable.

Also I think there are enums available for the "prs", "pass" and "fail" values you could use.

Reply:
If we leave as-is, there is just the question as to whether this method is actually only 'isTestStationWelsh' any more, and whether a different name would be appropriate?

Reply:
Splitting the logic crossed my mind as we may not have to mock everything for lookupPostcode in order to test the method. Will simplify the unit tests. As for the name we went back and forth a bit on functionality to capture, if we split the logic would be easy to have 'areFlagsEnabled' and 'isTestStationWelsh' separately